documentation/API_documentation/QoD_API.md

```diff
@@ -0,0 +1,178 @@
+
+The QoD API offers the application developers the capability to request for stable latency (reduced jitter) or throughput for some specified application data flows between application clients (within a User Equipment) and Application Servers (backend services). The developer has a pre-defined set of Quality of Service (QoS) Profiles which they could choose from depending on their latency or throughput requirements.
+
+<img src="./resources/QoD_overview.PNG" alt="QoD_LM" title="QoD API Overview">
+
+## 2\. Quick Start
+
+The usage of the API is based on QoS sessions, which can be created (based on available QoS profiles), queried and deleted.
+The deletion of a requested session can be triggered by the user or can be triggered automatically. The automatic process is triggered either when the user specified duration of a QoS session has reached its limit or the default session expiration time has been reached (within an example provider implementation it is set to 24hrs).
+
+Before starting to use the API, the developer needs to know about the below specified details:
+
+**Base-URL**
+The RESTful QoD API endpoint
+
+**Authentication**
+Security access keys such as OAuth 2.0 client credentials used by Client applications to invoke the QoD API.
+
+**QoS Profile**
+Latency or throughput requirements of the application mapped to relevant QoS profile class.
+
+**Identifier for the the user equipment (UE)**
+At least one identifier for the user equipment out of four options: IPv4 address, IPv6 address, MSISDN, or external ID [[5]](#5) assigned by the Mobile Operator for the user equipment
+
+**Identifier for the application server (AS)**
+IPv4 and/or IPv6 address of the application server (application backend)
+
+**App-Flow**
+The precise IP flow the developer wants to prioritize and have stable latency or throughput for. This flow is in the current API version determined by the identifiers used for the user equipment and the application server. And can be further elaborated with details such as ports or port-ranges. Future version of the API might allow more detailed flow descriptions.
+
+**Duration** 
+Number of seconds for which the QoD session should be created. This parameter is optional and if not specified, the session is either deleted on user request or if default expiration limit has been reached (24 hours in one of the example provider implementations).
+
+**Notification URL and token**
+Developers have a chance to specify callback URL on which notifications (eg. session termination) regarding the session can be received from the service provider. This is an optional parameter.
+
+Sample API invocations are presented in Section 4.6.
+
+## 3\. Authentication and Authorization
```

Comment:
We have to consider as well a three_legged securityScheme. I see it's not considered in the current 0.8 but it's used in our integrations. We can make a separate PR to add it explicitly.

Reply:
@jlurien, all: Aha, I missed, that the IPv6 was introduced. Still, we should not mix usage of IPv4 and IPv6 in the same QoD Session. Either ueId and AsId are IPv4 or both are IPv6.

Reply:
@tlohmar: See my comment above https://github.com/camaraproject/QualityOnDemand/pull/71#discussion_r1031883038